src/escalus_pubsub_stanza.erl

```diff
@@ -114,11 +112,14 @@ entry_body_with_sample_device_id_2() ->
 %% provide EntryBody as list of anything compliant with exml entity records.
 
 publish_entry_children([]) ->
-    entry_body_sample1();
+    entry_body_sample();
 
-publish_entry_children([#xmlel{}] = EntryBody) ->
+publish_entry_children([#xmlel{} | []] = EntryBody) ->
```

Comment:
This pattern is overly complex - it's _obvious already_ that we match on a one element long list, there's no point in adding `| []` - it only adds line noise.